frontend/packages/console-shared/src/test-utils/utils.ts

```diff
@@ -143,3 +143,16 @@ export const waitForStringNotInElement = (elem: ElementFinder, needle: string) =
     return !content.includes(needle);
   };
 };
+
+/**
+ * Search YAML manifest for a given string. Return true if found.
+ * @param     {string}    needle    String to search in YAML.
+ * @param     {string}    name      Name of the resource.
+ * @param     {string}    namespace Namespace of the resource.
+ * @param     {string}    kind      Kind of the resource.
+ * @returns   {boolean}             True if found, false otherwise.
+ */
+export function searchYAML(needle: string, name: string, namespace: string, kind: string): boolean {
```

Comment:
I know this is an old PR that's already merged, but this isn't the way we should test for content in the resources. Instead, we should

1. Get the JSON with `-o json`
2. Parse the JSON
3. Check the specific property we're interested in

The `searchYAML` approach is bad because it can result in incorrect failures due to different quoting styles in the YAML **or** it can result in false positives where the string is matched somewhere else in the document.

Can we look at removing this utility and replacing its usage? At the very least, I'd like to move this out of console-shared to discourage its use elsewhere.

@rhrazdil

Reply:
I agree, it's currently only used in old kubevirt tests, I'll update those and remove this method from console-shared